### PR TITLE
feat: DRY todo done, add .gitignore(python) for this repo

### PR DESCRIPTION
**Pull request by yihong0618**
This `PR` completes the **TODO**

https://github.com/Yikun/hub-mirror-action/blob/132b206a796926508b73fe82e9f3bb0969a3add6/hub-mirror/hub.py#L94

### CODE REVIEW

File: hub-mirror/hub.py [hunk 1]
```diff
@@ -86,29 +86,17 @@ def dynamic_list(self):
         return self._get_all_repo_names(url)
 
     @functools.lru_cache
-    def _get_all_repo_names(self, url):
-        page, per_page = 1, 60
-        api = url + "?page=0&per_page=" + str(per_page)
+    def _get_all_repo_names(self, url, page=1):
+        per_page = 60
+        api = url + f"?page={page}&per_page=" + str(per_page)
```
Comment (Yikun) [inline hub-mirror/hub.py:91]: A quick question: the page start is changed from `0` to `1`, start `0` and start`1`, have same result, right?

File: hub-mirror/hub.py [hunk 2]
```diff
@@ -86,29 +86,17 @@ def dynamic_list(self):
...
-                url_next = response.links['next']['url']
-                response = self.session.get(url_next)
-                # TODO: DRY
-                if response.status_code != 200:
-                    print("Repo getting failed: " + response.text)
-                    return []
-                page += 1
-                items = response.json()
-
+            return names + self._get_all_repo_names(url, page=page+1)
```
Comment (Yikun) [inline hub-mirror/hub.py:101]: nice recursion

**Comment by Yikun** [comment]
Thanks for your contribution, I do some basic test on https://github.com/Yikun/hub-mirror-action/pull/102 . It works.

The code is LGTM, I think it's ok to merge after the question anwsered.

**Comment by yihong0618** [comment]
> Thanks for your contribution, I do some basic test on #102 . It works.
> 
> The code is LGTM, I think it's ok to merge after the question anwsered.

But  I am not 100% sure, you can test by change the per page, starts from 1 seems more make sense.

**Comment by yihong0618** [comment]
@Yikun 

I search the GitHub API Doc, found it is from 1.

see: https://docs.github.com/cn/rest/guides/traversing-with-pagination

**Comment by Yikun** [comment]
恩，gitee和github都没问题：
github: [This makes sense, since by default, all paginated queries start at page 1.](https://docs.github.com/en/rest/guides/traversing-with-pagination#basics-of-pagination)
gitee: https://gitee.com/api/v5/swagger#/getV5UsersUsernameRepos

**Comment by yihong0618** [comment]
> 恩，gitee和github都没问题：
> github: [This makes sense, since by default, all paginated queries start at page 1.](https://docs.github.com/en/rest/guides/traversing-with-pagination#basics-of-pagination)
> gitee: https://gitee.com/api/v5/swagger#/getV5UsersUsernameRepos

哈哈，算是修了一个潜在的 bug~

**Comment by Yikun** [comment]
Merged, : )

**Comment by yihong0618** [comment]
> Merged, : )

Thanks.

**Comment by Yikun** [comment]
目前贡献者的CI仅跑了flake8，贡献体验不太好，在https://github.com/Yikun/hub-mirror-action/issues/103 跟踪一下，用例集成

**Pull request merged**